flink-formats/flink-compress/src/main/java/org/apache/flink/formats/compress/CompressWriterFactory.java

```diff
@@ -42,39 +47,57 @@
 
 	private Extractor<IN> extractor;
 	private CompressionCodec hadoopCodec;
+	private String hadoopCodecName;
+	private Map<String, String> hadoopConfigurationMap;
+	private String codecExtension;
```

Comment:
codecExtension? I believe so. Like when we have an instance for `CompressWriterFactory` created which precedes StreamingFileSink creation part so we can use this method to pass in the extension info to `OutputFileConfig` to set part suffix, etc. Let me know, if you think otherwise.

Reply:
No, I agree with you :)